lib/mongo/crypt/handle.rb

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) 2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'ffi'
+require 'base64'
+
+module Mongo
+  module Crypt
+
+    # A handle to the libmongocrypt library that wraps a mongocrypt_t object,
+    # allowing clients to set options on that object or perform operations such
+    # as encryption and decryption
+    class Handle
+
+      # Creates a new Handle object and initializes it with options
+      #
+      # @example Instantiate a Handle object with local KMS provider options
+      #   Mongo::Crypt::Handle.new({
+      #     local: { key: 'MASTER-KEY' }
+      #   })
+      #
+      # @param [ Hash ] kms_providers A hash of KMS settings. The only supported key
+      # is currently :local. Local KMS options must be passed in the format
+      # { local: { key: 'MASTER-KEY' } } where the master key is a 96-byte, base64
+      # encoded string.
+      #
+      # @since 2.12.0
+      def initialize(kms_providers)
```

Comment:
There will be one or two more params here, but they're not required for explicit encryption.